feature/experimental/gribi/ate_tests/dut_daemon_failure/dut_daemon_failure_test.go

```diff
@@ -0,0 +1,393 @@
+
+	return flow
+}
+
+// stopAndVerifyTraffic stops traffic on the ATE
+// and checks for packet loss for the given flow.
+func stopAndVerifyTraffic(t *testing.T, ate *ondatra.ATEDevice, flow *ondatra.Flow) {
+
+	ate.Traffic().Stop(t)
+
+	time.Sleep(time.Minute)
+
+	flowPath := ate.Telemetry().Flow(flow.Name())
+	if got := flowPath.LossPct().Get(t); got != 0 {
+		t.Errorf("FAIL: LossPct for flow named %s got %g, want 0", flow.Name(), got)
+	} else {
+		t.Logf("LossPct for flow named %s got %g, want 0", flow.Name(), got)
+	}
+}
+
+// testArgs holds the objects needed by the test case.
+type testArgs struct {
+	ctx context.Context
+	dut *ondatra.DUTDevice
+	ate *ondatra.ATEDevice
+	top *ondatra.ATETopology
+}
+
+// addRoute programs an IPv4 route entry through the given gRIBI client
+// after programming the necessary nexthop and nexthop-group.
+func addRoute(ctx context.Context, t *testing.T, args *testArgs, clientA *gribi.Client) {
+	t.Logf("Add an IPv4Entry for %s pointing to ATE port-2 via clientA", ateDstNetCIDR)
+	clientA.AddNH(t, nhIndex, atePort2.IPv4, *deviations.DefaultNetworkInstance, fluent.InstalledInRIB)
+	clientA.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInRIB)
+	clientA.AddIPv4(t, ateDstNetCIDR, nhgIndex, *deviations.DefaultNetworkInstance, "", fluent.InstalledInRIB)
+}
+
+// verifyAFT verifies through AFT Telemetry if a route is present on the DUT.
+func verifyAFT(ctx context.Context, t *testing.T, args *testArgs) {
+	t.Logf("Verify through AFT Telemetry that %s is active", ateDstNetCIDR)
+	ipv4Path := args.dut.Telemetry().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, ok := ipv4Path.Prefix().Watch(t, time.Minute, func(val *telemetry.QualifiedString) bool {
+		return val.IsPresent() && val.Val(t) == ateDstNetCIDR
+	}).Await(t); !ok {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got.Val(t), ateDstNetCIDR)
+	}
+}
+
+// verifyTraffic verifies that traffic flows through the DUT without any loss.
+func verifyTraffic(ctx context.Context, t *testing.T, args *testArgs) {
+	t.Logf("Verify by running traffic that %s is active", ateDstNetCIDR)
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
+
+	flow := startTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint)
+	t.Logf("Wait for 15 seconds")
+	time.Sleep(15 * time.Second)
+	stopAndVerifyTraffic(t, args.ate, flow)
+
+}
+
+// verifyGribiGet verifies through gRIBI Get RPC if a route is present on the DUT.
+func verifyGribiGet(ctx context.Context, t *testing.T, clientA *gribi.Client) {
+	t.Logf("Verify through gRIBI Get RPC that %s is present", ateDstNetCIDR)
+	getResponse, err := clientA.Fluent(t).Get().WithNetworkInstance(*deviations.DefaultNetworkInstance).WithAFT(fluent.IPv4).Send()
+	if err != nil {
+		t.Errorf("Cannot Get: %v", err)
+	}
+	entries := getResponse.GetEntry()
+	var found bool = false
```

Comment:
The zero value of a bool is already false.

```suggestion
	var found bool
```

Or if you want to set it explicitly to false, we don't need to say that it's a bool: `found := false`.